indra/llcommon/llcallbacklist.h

```diff
@@ -27,53 +27,248 @@
+/*-------------------- legacy names in global namespace --------------------*/
+#define gIdleCallbacks (LLCallbackList::instance())
+
+using nullary_func_t = LLCallbackList::callable_t;
+using bool_func_t    = LLCallbackList::bool_func_t;
 
 // Call a given callable once in idle loop.
-void doOnIdleOneTime(nullary_func_t callable);
+inline
+LLCallbackList::handle_t doOnIdleOneTime(nullary_func_t callable)
+{
+    return gIdleCallbacks.doOnIdleOneTime(callable);
+}
 
 // Repeatedly call a callable in idle loop until it returns true.
-void doOnIdleRepeating(bool_func_t callable);
+inline
+LLCallbackList::handle_t doOnIdleRepeating(bool_func_t callable)
+{
+    return gIdleCallbacks.doOnIdleRepeating(callable);
+}
+
+/*****************************************************************************
+*   LLLater: callbacks at some future time
+*****************************************************************************/
+class LLLater: public LLSingleton<LLLater>
+{
+    LLSINGLETON(LLLater);
+
+    using token_t = U32;
+
+    // Define a struct for our priority queue entries, instead of using
+    // a tuple, because we need to define the comparison operator.
+    struct func_at
+    {
+        nullary_func_t mFunc;
+        token_t mToken;
+        LLDate::timestamp mTime;
+
+        func_at(const nullary_func_t& func, token_t token, LLDate::timestamp tm):
+            mFunc(func),
+            mToken(token),
+            mTime(tm)
+        {}
+
+        friend bool operator<(const func_at& lhs, const func_at& rhs)
+        {
+            // use greater-than because we want fibonacci_heap to select the
+            // EARLIEST time as the top()
+            return lhs.mTime > rhs.mTime;
+        }
+    };
+
+    // Accept default stable<false>: when two funcs have the same timestamp,
+    // we don't care in what order they're called.
+    // Specify constant_time_size<false>: we don't need to optimize the size()
+    // method, iow we don't need to store and maintain a count of entries.
+    typedef boost::heap::fibonacci_heap<func_at, boost::heap::constant_time_size<false>>
+        queue_t;
+
+public:
+    // If tasks that come ready during a given tick() take longer than this,
+    // defer any subsequent ready tasks to a future tick() call.
+    static constexpr F32 TIMESLICE{ 0.005f };
+
+    class handle_t
+    {
+    private:
+        friend class LLLater;
+        token_t token;
+    public:
+        handle_t(token_t token=0): token(token) {}
+        bool operator==(const handle_t& rhs) const { return this->token == rhs.token; }
+        explicit operator bool() const { return bool(token); }
+        bool operator!() const { return ! bool(*this); }
+    };
 
+    // Call a given callable once at specified timestamp.
+    handle_t doAtTime(nullary_func_t callable, LLDate::timestamp time);
+
+    // Call a given callable once after specified interval.
+    handle_t doAfterInterval(nullary_func_t callable, F32 seconds);
```

Comment:
Here I'd propose `scheduleAfter`

Plus create a new type `seconds`. [strong type iz strong!](https://www.fluentcpp.com/2016/12/08/strong-types-for-strong-interfaces/)

This comes with the downside of making it impossible to mix F32 and seconds at will, but with the upside that they cannot just be mixed and misinterpreted

Reply:
Hmm, tough spot. I suppose the questions being how much time it would consume to change that or if it is cheaper as in work and later maintenance to follow bad principles

Reply:
Specifically, I want to make it as painless as possible for other maintainers to say, "Hmm, you know, this `LLEventTimer` subclass is way too much boilerplate for this simple use case" and drop in `LL::Timers` (or whatever) instead.